File: 1. Learn the Basics/1. Pattern/Python/22.py
# 4444444
# 4333334
# 4322234
# 4321234
# 4322234
# 4333334
# 4444444


def getNumberPattern(n):
    for i in range(0,2*n-1):
        for j in range(0,2*n-1):
            top = i
            left = j
            bottom = (2*n-2)-i
            right = (2*n-2)-j
            print(n-min(min(top,bottom),min(left,right)),end="")
        print("\r")

n = int(input())
getNumberPattern(n)

File: 1. Learn the Basics/1. Pattern/C++/22.cpp
/*

4444444
4333334
4322234
4321234
4322234
4333334
4444444

*/

#include <bits/stdc++.h>
using namespace std;

void getNumberPattern(int n) {
    // Write your code here.
    for(int i=0;i<2*n-1;i++){
        for(int j=0;j<2*n-1;j++){
            int top = i;
            int left = j;
            int bottom = (2*n-2)-i;
            int right = (2*n-2)-j;
            cout << (n-min(min(top, bottom),min(left, right)));
        }
        cout << endl;
    }
}


int main() {
	// your code goes here
	int n;
	cin >> n;
	getNumberPattern(n);
	return 0;
}
